core/cap-0046-06.md

```diff
@@ -76,136 +76,77 @@ type describes the way in which the type is serialized as an `SCVal`.
-//     AccountAuthorization, // authorization
-// ]
-struct KeyedAccountAuthorization {
-    u256 publicKey;
-    AccountAuthorization authorization;
+pub struct AccountSignatures {
+    pub account_id: AccountId,
+    pub signatures: Vec,
 }
 
-// SCO_VEC[
-//     0,                    // Authorization::Contract, 4 bytes
-// ] | SCO_VEC[
-//     1,                    // Authorization::Ed25519, 4 bytes
-//     Ed25519Authorization,
-// ] | SCO_VEC[
-//     2,                    // Authorization::Account, 4 bytes
-//     AccountAuthorization,
-// ]
-enum Authorization {
-    Contract,
-    Ed25519(Ed25519Authorization),
-    Account(AccountAuthorization),
+pub enum Signature {
+    Invoker,
+    Ed25519(Ed25519Signature),
+    Account(AccountSignatures),
 }
 
-// SCO_VEC[
-//     0,                         // KeyedAuthorization::Contract, 4 bytes
-// ] | SCO_VEC[
-//     1,                         // KeyedAuthorization::Ed25519, 4 bytes
-//     KeyedEd25519Authorization,
-// ] | SCO_VEC[
-//     2,                         // KeyedAuthorization::Account, 4 bytes
-//     KeyedAccountAuthorization,
-// ]
-enum KeyedAuthorization {
-    Contract,
-    Ed25519(KeyedEd25519Authorization),
-    Account(KeyedAccountAuthorization),
+pub enum Identifier {
+    Contract(BytesN<32>),
+    Ed25519(BytesN<32>),
+    Account(AccountId),
 }
 
-// SCO_VEC[
-//     SCV_U32,    // {0, 1, 2}
-//     SCO_BINARY, // 32 bytes
-// ]
-enum Identifier {
-    Contract(u256),
-    Ed25519(u256),
-    Account(u256),
-}
+// Byte arrays and strings can be passed to contracts and stores using the Bytes type.
+// BytesN is fixed size, while Bytes is variable
+pub struct Bytes(EnvVal<Host, Object>);
+pub struct BytesN<const N: usize>(EnvVal<Host, Object>);
 ```
 
 ### Semantics: Signatures
 
-Signatures for this contract are over the following `Message` type
+Signatures for this contract are over the following `SignaturePayload` type
 
 ```rust
-// SCO_VEC[
-//     0,          // Caller::Transaction, 4 bytes
-// ] | SCO_VEC[
-//     1,          // Caller::Contract, 4 bytes
-//     SCO_BINARY, // 32 bytes
-// ]
-enum Caller {
-    Transaction,
-    Contract(u256),
-}
 
-// SCO_VEC[
-//     SCO_BINARY, // nonce, 32 bytes
-//     SCO_BINARY, // thisContract, 32 bytes
-//     SCV_SYMBOL, // symbol
-//     SCO_VEC[    // parameters
-//         SCVal,
-//         ...,
-//     ],
-//     Caller,     // caller
-// ]
-struct MessageV0 {
-    u256 nonce;
-    u256 thisContract;
-    SCSymbol symbol;
-    Vec<SCVal> parameters;
-    Caller caller;
+pub struct SignaturePayloadV0 {
+    pub network: Bytes,
+    pub contract: BytesN<32>,
+    pub name: Symbol,
+    pub args: Vec,
 }
 
-// SCO_VEC[
-//     0,         // Message::V0, 4 bytes
-//     MessageV0,
-// ]
-enum Message {
-    V0(MessageV0),
+pub enum SignaturePayload {
+    V0(SignaturePayloadV0),
 }
+
+```
+
+### Semantics: Initialization
+
+There are two ways to initialize the token contract
```

Comment:
I think this is a bit misleading. `init_asset` is not exposed anywhere. I think CAP should refer to the interfaces, i.e. host functions in the env (though token contract creation is in CAP-46-02...) and the token spec interface (this one has just one `init` method; also host shouldn't be mentioned in the interface, similar to the following methods)